lib/hpke/src/certificate.cpp

```diff
@@ -59,8 +180,23 @@ struct Certificate::ParsedCertificate
     return make_typed_unique<EVP_PKEY>(X509_get_pubkey(x509.get()));
   }
 
+  bool is_ca() const
```

Comment:
If you're going to parse and store everything else at instantiation time, you might as well do this as well.